### PR TITLE
Add Format Step into Build Pipeline

### PR DESCRIPTION
**Pull request by yannlabou**
Initial proposal (Travis-only)
We still need to define and commit a .clang-format file for the project
https://clang.llvm.org/docs/ClangFormat.html

### CODE REVIEW

**Comment by ihhub** [comment]
I have 2 things to clarify:
1. We need to add a new `.clang-format` file  with needed code style settings before committing these changes. Is it correct?
2. I am proposing to merge new Travis-CI job with already existed `STATIC_ANALYSIS` job as they are similar (in some sense of analysis).

**Comment by yannlabou** [comment]
> 1. We need to add a new `.clang-format` file  with needed code style settings before committing these changes. Is it correct?

Not exactly. We can write our custom .clang-format or use a predefined style. In my pr, I'm using the [llvm style](https://github.com/ihhub/penguinV/pull/379/files#diff-1969a46bbf0a406a157860f6aaf08b47R28) (`-style=llvm`). Possible [options](http://releases.llvm.org/3.8.0/tools/clang/docs/ClangFormatStyleOptions.html#configurable-format-style-options) are 
- LLVM
- Google
- Chromium
- Mozilla
- WebKit

**Comment by ihhub** [comment]
@yannlabou my apology is I ask a very stupid question but where is an output of the format check?
I only see this:
```
Checking clang-format between TRAVIS_BRANCH=master and TRAVIS_PULL_REQUEST_BRANCH=issue-280-add-format-step-into-build-pipeline
No source code to check for formatting.
```

**Comment by ihhub** [comment]
Ah, it checks only for difference changes. I don't have any more questions (we need to test it). Is it a final version of the change?

File: script/check_code_format.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Script to determine if source code in Pull Request is properly formatted.
+# Exits with non 0 exit code if formatting is needed.
+#
+# This script assumes to be invoked at the project root directory.
+
+FILES_TO_CHECK=$(git diff --name-only HEAD^ | grep -E ".*\.(cpp|cc|c\+\+|cxx|c|h|hpp)$")
```
Comment (ihhub) [inline script/check_code_format.sh]: We miss `cu` and `cuh` files (for CUDA).

**Comment by ihhub** [comment]
Shall we create a separate branch to test such and then set proper settings for code analysis?

**Comment by yannlabou** [comment]
> Shall we create a separate branch to test such and then set proper settings for code analysis?

It’s a good idea as we need to modify a lot of files to see all the effects of he pull request.

File: script/check_code_format.sh (hunk 1 above)
Comment (yannlabou) [reply script/check_code_format.sh]: Sure.

**Comment by yannlabou** [comment]
> Ah, it checks only for difference changes. I don't have any more questions (we need to test it). Is it a final version of the change?

Yes it is. Now we need to test on a couple of sources.

**Comment by ihhub** [comment]
Okay. I will create a new branch today (probably in 5-6 hours time).

**Comment by ihhub** [comment]
I've created a new branch called **code_style**. Could you please create a pull request for this branch?

**Comment by ihhub** [comment]
Closing this pull request as we work on different branch,

**Pull request closed without merging**